torch/csrc/autograd/custom_function.cpp

```diff
@@ -309,9 +309,13 @@ static optional_variable_list _process_backward_mode_ad(
       }
       // No need to mark as modified Tensors that are not inputs.
       if (!is_input) {
-        TORCH_WARN(
-            "Only input Tensors should be given to ctx.mark_dirty(). If a Tensor is not an input, there"
-            " is no need to pass it to mark_dirty().");
+        const char* mark_dirty_error_msg =
+            "ctx.mark_dirty() received a tensor that was not an input. "
+            "Only input Tensors that have been mutated should be passed to "
+            "ctx.mark_dirty().";
+        // We reach this path in the view of intermediate case
+        TORCH_CHECK(!var.is_view(), mark_dirty_error_msg);
```

Comment:
How does the intermediate case result in var being a view?

Reply:
a view of a intermediate tensor is still a view, but unlike doing a view of input, the view of the intermediate base does not require grad because the intermediate base does not require grad

Reply:
What prevents an input from being a view?

Reply:
Inputs can be views and that is fine, we should do the usual rebase history stuff. If you mean to ask what happens if you are a view of the input, your view will be a leaf that require_grad being modified inplace and that is handled else-where (the previous branch of the if-else).

This PR is mainly targeting the case where the view was created from an intermediate, in which we are leaf with requires_grad=False and will immediately error (with an internal assert) because this autograd.Function is marked dirty and will call rebase_history on it.